tests/test_core/test_alignment.py

```diff
@@ -2572,17 +2572,29 @@ def test_add_from_ref_aln(self):
         )  # test wrong_refseq
 
     def test_annotate_matches_to(self):
-        """Tests annotate_matches_to method in Aligned class."""
-        if isinstance(self, Sequence):
-            seq = self.SEQ("CCCCCAAAGTACCCCCCAAAGTA", name="x")
-            pattern = "AAAGTA"
-            annot = seq.annotate_matches_to(
-                pattern=pattern, annot_type="domain", name="fred", allow_multiple=True
-            )
-            fred = annot.get_slice()
-            self.assertEqual(str(fred)[0 : len(pattern)], pattern)
+        """annotate_matches_to method should attach
+         annotations correctly to a Sequence object."""
+        seq = Sequence("TTCCACTTCCGCTT", name="x")
+        allow_multiple = True
+        if not isinstance(seq, Sequence):
+            return True
+        pattern = "CCRC"
+        regular_expression = DNA.to_regex(seq=pattern)
+        annot = seq.annotate_matches_to(
+            pattern=regular_expression,
+            annot_type="domain",
+            name="fred",
+            allow_multiple=allow_multiple,
+        )
+        if allow_multiple == True:
+            for i in range(0, len(annot)):
```

Comment:
loop not required, suggest [a.get_slice() for a in seq.annotate_matches_to()] == ["CCAC", "CCGC"]